internal/encryption/method/aesgcm/aesgcm.go

```diff
@@ -9,53 +9,129 @@ import (
 	"crypto/aes"
 	"crypto/cipher"
 	"crypto/rand"
+	"errors"
+
+	"github.com/opentofu/opentofu/internal/errorhandling"
+
+	"github.com/opentofu/opentofu/internal/encryption/method"
+	// This unsafe is required for go:linkname
+	_ "unsafe"
 )
 
+// Note: the linking below is a workaround for Go issue #42470. We want to support setting both the nonce and the tag
+// size specifically because if the defaults for any of these parameters change due to new cryptography research,
```

Comment:
I don't think this is likely to happen, fwiw. 

Tag size is limited at most at 16 bytes, which is the default IIRC. There's no reason to shorten the tag for this use case.

Nonce size also shouldn't change as I discuss elsewhere. I'd simplify this to the default `NewGCM` call. 

If you do anything, I'd consider https://github.com/opentofu/opentofu/pull/1291/files#r1499974976 instead.